channelserver/src/meta.rs

```diff
@@ -264,7 +345,8 @@ impl From<HttpRequest<WsChannelSessionState>> for SenderData {
         };
         // parse user-header for platform info
         sender.ua = get_ua(&headers, Some(&log));
-        sender.remote = req.connection_info().remote().map(|rem| rem.to_owned());
+        // Ideally, this would just get &req. For testing, I'm passing in the values.
+        sender.remote = get_remote(&req.headers(), &req.state().proxy_allowlist);
```

Comment:
I still maintain that we have to pass in `req.peer_addr()` as an argument to this function in order to make a correct decision; we shouldn't trust the `X-Forwarded-For` header unless we can verify that the connecting client is one of the trusted proxies.

Reply:
Fair point. added.